diff --git a/third_party/blink/renderer/core/frame/web_remote_frame_impl.cc b/third_party/blink/renderer/core/frame/web_remote_frame_impl.cc
--- a/third_party/blink/renderer/core/frame/web_remote_frame_impl.cc
+++ b/third_party/blink/renderer/core/frame/web_remote_frame_impl.cc
@@ -362,6 +362,8 @@
   visual_properties.is_pinch_gesture_active =
       ancestor_widget->PinchGestureActiveInMainFrame();
   visual_properties.screen_infos = ancestor_widget->GetOriginalScreenInfos();
+  visual_properties.emulated_screen_infos =
+      ancestor_widget->GetEmulatedScreenInfos();
   visual_properties.visible_viewport_size =
       ancestor_widget->VisibleViewportSize();
   const std::vector<gfx::Rect>& viewport_segments =
