diff --git a/content/browser/renderer_host/render_process_host_impl.cc b/content/browser/renderer_host/render_process_host_impl.cc
index a022e1d19be7905a7d31663e315d5941c069c605..7eedc420943dcb7d97cd8f229170aacd6a07ea66 100644
--- a/content/browser/renderer_host/render_process_host_impl.cc
+++ b/content/browser/renderer_host/render_process_host_impl.cc
@@ -2055,7 +2055,7 @@
          base::CommandLine::ForCurrentProcess()->GetSwitches()) {
       if (sw.first.compare(0, 4, "uxr-") == 0) {
         uxr_cfg[sw.first] =
-            base::CommandLine::ForCurrentProcess()->GetSwitchValueASCII(sw.first);
+            base::CommandLine::ForCurrentProcess()->GetSwitchValueUTF8(sw.first);
       }
     }
     // Snapshot construction is browser-owned and launch-scoped. Even an empty
