diff --git a/content/browser/renderer_host/render_widget_host_impl.h b/content/browser/renderer_host/render_widget_host_impl.h
--- a/content/browser/renderer_host/render_widget_host_impl.h
+++ b/content/browser/renderer_host/render_widget_host_impl.h
@@ -772,7 +772,9 @@
       bool is_pinch_gesture_active,
       const gfx::Size& visible_viewport_size,
       const gfx::Rect& compositor_viewport,
-      std::vector<gfx::Rect> root_widget_viewport_segments);
+      std::vector<gfx::Rect> root_widget_viewport_segments,
+      const std::optional<display::ScreenInfos>& emulated_screen_infos =
+          std::nullopt);
 
   // Indicates if the render widget host should track the render widget's size
   // as opposed to visa versa.
@@ -1443,6 +1445,8 @@
     MainFramePropagationProperties();
     ~MainFramePropagationProperties();
 
+    std::optional<display::ScreenInfos> emulated_screen_infos;
+
     // The page-scale factor of the main-frame.
     float page_scale_factor = 1.f;
 
